### PR TITLE
pass config option to mr.bob

### LINKED ISSUE #80: Add "--config" option to plonecli "create" and "add" commands
It would be very helpful if `plonecli` could accept a configuration file to specify mr.bob variables.

This could be easily be done in the functions `create` and `add` in `plonecli.cli.py` by adding a `click.option` and extending the list passed to `mrbobmain` as follows:

https://github.com/plone/plonecli/blob/d49d7c79613ac3324e85da45c5a8daeb09cd0a1f/plonecli/cli.py#L77

```python
...
@click.option("-c", "--config", default=None, help="Configuration file to specify either [mr.bob] or [variables] sections.")
...
def create(context, template, name, config):
    ...
    mrbob_args = [bobtemplate, "-O", name]
    if config is not None:
        mrbob_args.extend(["-c", config])
    mrbobmain(mrbob_args)

```

https://github.com/plone/plonecli/blob/d49d7c79613ac3324e85da45c5a8daeb09cd0a1f/plonecli/cli.py#L97

```python
@click.option("-c", "--config", default=None, help="Configuration file to specify either [mr.bob] or [variables] sections.")
...
def add(context, template, config):
    ...
    mrbob_args = [bobtemplate]
    if config is not None:
        mrbob_args.extend(["-c", config])
    mrbobmain(mrbob_args)
```

### PR DESCRIPTION
**Pull request by erral**
Fixes #80 

@me-kell

### CODE REVIEW

**Comment by me-kell** [comment]
@erral: Thanks!

**Review by MrTango — CHANGES_REQUESTED** [review]
Now i get what this is about.
We can point mrbob to another config file other than the one in the user home right?
That sounds useful, but i would not name it config, because we already have a config command which will lead to confusions. So something like `bobconfig` would be better.
And we should make it clear in the help strings too, that this is about mrbob config.

File: CHANGES.rst [hunk 1]
```diff
@@ -4,8 +4,8 @@ History
 2.5 (unreleased)
 ----------------
 
-- Nothing changed yet.
-
+- Provide a way to pass a configuration file to create and add commands
```
Comment (MrTango) [inline CHANGES.rst:7]:
```suggestion
- Provide a way to pass a mrbob configuration file
- ``

File: plonecli/cli.py [hunk 2]
```diff
@@ -73,8 +73,14 @@ def cli(context, list_templates, versions):
 @cli.command()
 @click.argument("template", type=click.STRING, shell_complete=get_templates)
 @click.argument("name")
+@click.option(
+    "-c",
+    "--config",
+    default=None,
+    help="Configuration file to specify either [mr.bob] or [variables] sections.",
```
Comment (MrTango) [inline plonecli/cli.py:80]:
```suggestion
    help="mrbob configuration file. The default is ~/.mrbob.",
```

File: plonecli/cli.py [hunk 3]
```diff
@@ -83,18 +89,29 @@ def create(context, template, name):
...
+    mrbobmain(mrbob_args)
 
 
 @cli.command()
 @click.argument("template", type=click.STRING, shell_complete=get_templates)
+@click.option(
+    "-c",
+    "--config",
+    default=None,
+    help="Configuration file to specify either [mr.bob] or [variables] sections.",
```
Comment (MrTango) [inline plonecli/cli.py:111]:
```suggestion
    help="mrbob configuration file. The default is ~/.mrbob.",
```

**Comment by erral** [comment]
I have commited your suggestions and changed also the option and variable names to bobconfig.

With that it should be clear that is a specific mr.bob config  file.

File: CHANGES.rst [hunk 4]
```diff
@@ -4,8 +4,9 @@ History
 2.5 (unreleased)
 ----------------
 
-- Nothing changed yet.
-
+- Provide a way to pass a mrbob configuration file
+ ``
```
Comment (jensens) [inline CHANGES.rst:8]: are these two backticks here by intend or accident?

Comment (erral) [reply CHANGES.rst:8]: It's just an accident.

**Comment by jensens** [comment]
@MrTango I think this is good for merge, or do you think something is missing?

**Review by MrTango — APPROVED** [review]

**Pull request merged**